kratos/modeler/copy_properties_modeler.cpp

```diff
@@ -0,0 +1,89 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "modeler/copy_properties_modeler.h"
+#include "utilities/parallel_utilities.h"
+
+namespace Kratos
+{
+
+CopyPropertiesModeler::CopyPropertiesModeler(
+    Model& rModel,
+    Parameters ModelerParameters)
+        : Modeler(rModel, ModelerParameters)
+        , mpModel(&rModel)
+{
+    mParameters.ValidateAndAssignDefaults(this->GetDefaultParameters());
+}
+
+Modeler::Pointer CopyPropertiesModeler::Create(Model& rModel,
+    const Parameters ModelParameters) const
+{
+    return Kratos::make_shared<CopyPropertiesModeler>(rModel, ModelParameters);
+}
+
+const Parameters CopyPropertiesModeler::GetDefaultParameters() const
+{
+    const Parameters default_parameters = Parameters(R"({
+        "echo_level"                  : 0,
+        "origin_model_part_name"      : "",
+        "destination_model_part_name" : ""
+    })");
+    return default_parameters;
+}
+
+void CopyPropertiesModeler::SetupModelPart()
+{
+    const auto origin_model_part_name = mParameters["origin_model_part_name"].GetString();
+    const auto destination_model_part_name = mParameters["destination_model_part_name"].GetString();
+    ModelPart& r_origin_model_part = mpModel->GetModelPart(origin_model_part_name);
+    ModelPart& r_destination_model_part = mpModel->GetModelPart(destination_model_part_name);
```

Comment:
```suggestion
    const std::string origin_model_part_name = mParameters["origin_model_part_name"].GetString();
    const std::string destination_model_part_name = mParameters["destination_model_part_name"].GetString();
    const auto& r_origin_model_part = mpModel->GetModelPart(origin_model_part_name);
    auto& r_destination_model_part = mpModel->GetModelPart(destination_model_part_name);
```
We tend to avoid using `auto` with fundamental types. On the contrary we tend to use it with custom (potentially more complex and prone to change) types. Besides I think that the origin model part reference can be `const`.

Reply:
> Besides I think that the origin model part reference can be `const`

Unfortunately, the method `rProperties` is not const